daemon/cmd/daemon.go

```diff
@@ -1192,7 +1192,15 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup,
 			node.GetInternalIPv4Router(), node.GetIPv6Router())
 		if err != nil {
 			log.WithError(err).Warning("Cannot annotate k8s node with CIDR range")
+		} else {
+			cleaner.cleanupFuncs.Add(func() {
```

Comment:
We should only do this if, as mentioned in the original issue, clean-cilium-state is set.

Reply:
For now, I removed this part and only pushed refactoring + new function for removing node annotations. I will work on this in separate PR.